doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,246 @@
+
+#### Story 2
+
+Scorecard will include an example custom test that end users could use as a reference when writing their custom tests.
+
+#### Story 3
+
+Scorecard internal tests will be converted to the custom test format and therefore be externalized from the scorecard binary itself.
+
+#### Story 4
+
+Scorecard would process tests based on an on-disk directory or bundle image that contains the scorecard configuration.
+
+#### Story 5 
+
+- I am as an Operator developer, would like to be able to create tests using the same Assert syntax adopted by [Kuttl](https://github.com/kudobuilder/kuttl) and check its results aggregated within scorecard results. 
+
+**Implementation Notes**
+
+The Scorecard would support the creation of a Kuttl test image and the execute tests via Kuttl, however, it would be transparent for users since its results will be integrated into scorecard results.
+
+## Design Details
+
+### Glossary
+
+ * test image - this is a container image built by end-users to contain a custom test implementation
+ * test configuration (DSL) - this is a file that contains the test definitions and labels which determine what tests are run and how they are configured
+ * Custom Test Definition - A custom test is any sort of test logic required, written in any programming language that satifies the scorecard test requirements for output format and execution.
+
+#### Test Image
+
+A custom test is based on a container image, that image is created by
+the end-user using whatever tools they want to use.
+
+The test image must produce container log output that conforms to the scorecard Test Result output.  The output is an array of Test Results, so a test image could contain multiple tests if necessary.  Here is a sample of test output in the required format:
+```
+    [
+      {
+	      "name": "Spec fields with descriptors",
+	      "description": "All spec fields have matching descriptors in the CSV",
+	      "state": "fail",
+	      "suggestions": [
+		"Add a spec descriptor for size"
+	      ],
+	      "crname": "example-memcached"
+      }, 
+      {
+	      "name": "Some other test",
+	      "description": "Another test on the CSV",
+	      "state": "pass",
+	      "suggestions": [
+	      ],
+	      "crname": "example-memcached"
+      },
+    ]
+```
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
+
+Also, for each test execution, scorecard would also make the
+bundle image contents available to test images by mounting
+the bundle image contents into the following location inside the test pod:
+
+```
+/scorecard
+```
+
+This allows tests to have access to all the bundle contents if they 
+are required for the test logic.  Note:  It is the responsibility of the test to iterate though each CR that they want to test and create Custom Resources as scorecard will no longer be responsible for this.  The list of CRs to test are found in the CSV within the bundle image content.
```

Comment:
We're not really making any assumptions about what the test does. It could:
1. Not use the CRs at all (e.g. doing a static bundle validation)
2. Use the CRs in the CSV's `alm-examples` annotation (e.g. to verify they validate the associated CRD schema)
3. Use some other CR(s) built (or maybe mounted) into the test image.

Reply:
I added this almost verbatim in latest commit.  Thanks!